hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -511,22 +514,33 @@ ContainerCommandResponseProto> executePutBlock(boolean close,
           putBlockAsync(xceiverClient, blockData, close, tokenString);
       CompletableFuture<ContainerProtos.ContainerCommandResponseProto> future =
           asyncReply.getResponse();
+
       flushFuture = future.thenApplyAsync(e -> {
         try {
           validateResponse(e);
         } catch (IOException sce) {
           throw new CompletionException(sce);
         }
+        boolean timedOut = e.getMessage().equals("TIMEOUT");
```

Comment:
hacky. Is there a better way to detect timeout?

Reply:
This is a workaround to avoid further refactoring. I just need to get the message across without breaking other parts.

Reply:
A better way to do this might be to add a new field to the proto message. I will consider this later.